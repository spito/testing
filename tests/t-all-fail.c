#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>

#include <cut.h>

TEST(prvni) {
    ASSERT(1);
}

TEST(druhy) {
    SUBTEST(sub1) {
        ASSERT(1);
    }
    SUBTEST(sub2) {
        ASSERT(0);
    }
    SUBTEST(sub3) {
        ASSERT(1);
    }
}

TEST(treti) {
    ASSERT(0);
}

TEST(ctvrty) {
    DEBUG_MSG("co to je? asi %d", 6);
    int *i = 0;
    DEBUG_MSG("tohle jeste uvidim");
    *i = 6;
    DEBUG_MSG("ale tohle uz ne :(");
}

TEST(paty, TEST_SETTINGS(TEST_TIMEOUT(1))) {
    while(1);
}


TEST(sesty) {
    printf("cislo %d", 6);

    ASSERT_FILE(stdout, "cislo 6");
}

TEST(osm) {
    DEBUG_MSG("sedmy test neni");
    exit(10);
}
