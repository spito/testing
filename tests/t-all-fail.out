[  1] prvni..................................................................OK
[  2] druhy: 3 subtests
    sub1.....................................................................OK
    sub2...................................................................FAIL
        assert '0' (all-fail.c:16)

    sub3.....................................................................OK
[  2] druhy (overall)......................................................FAIL
[  3] treti................................................................FAIL
    assert '0' (all-fail.c:24)

[  4] ctvrty..........................................................SIGNALLED
    signal: SIGSEGV (11)
    debug messages:
      co to je? asi 6 (all-fail.c:28)
      tohle jeste uvidim (all-fail.c:30)

[  5] paty............................................................TIMED OUT
    timed out (1 s)

[  6] sesty..................................................................OK
[  7] osm..............................................................NON ZERO
    return code: 10
    debug messages:
      sedmy test neni (all-fail.c:47)


Summary:
  tests:          7
  succeeded:      2
  filtered out:   0
  suppressed:     0
  skipped:        0
  failed:         5

Points:           0.00
